cmd/warm-backend-s3.go

```diff
@@ -114,14 +115,37 @@ func newWarmBackendS3(conf madmin.TierS3, tier string) (*warmBackendS3, error) {
 		return nil, err
 	}
 	var creds *credentials.Credentials
-	if conf.AWSRole {
+	switch {
+	case conf.AWSRole:
 		creds = credentials.New(&credentials.IAM{
 			Client: &http.Client{
 				Transport: NewHTTPTransport(),
 			},
 		})
-	} else {
+	case conf.AWSRoleWebIdentityTokenFile != "" && conf.AWSRoleARN != "":
+		sessionName := conf.AWSRoleSessionName
+		if sessionName == "" {
+			sessionName = mustGetUUID()
```

Comment:
```suggestion
			sessionName = "minio-tier-"-randString(10)
```

Reply:
let's do 'minio-tier-'+mustGetUUID(); that way it will never break; AWS role name chars are limited:

```
The regex used to validate this parameter (RoleSessionName) is a string of characters consisting of upper- and lower-case alphanumeric characters with no spaces. You can also include underscores or any of the following characters: =,.@-
```